lib/rules/jsx-space-before-closing.js

```diff
@@ -63,6 +64,21 @@ module.exports = {
           return;
         }
 
+        if (
+          closingSlash.type !== 'Boolean'
+          && closingSlash.type !== 'Null'
+          && closingSlash.type !== 'Identifier'
+          && closingSlash.type !== 'Keyword'
+          && closingSlash.type !== 'Punctuator'
+          && closingSlash.type !== 'JSXIdentifier'
+          && closingSlash.type !== 'JSXText'
+          && closingSlash.type !== 'Numeric'
+          && closingSlash.type !== 'String'
+          && closingSlash.type !== 'RegularExpression'
+        ) {
+          return;
+        }
```

Comment:
tests?

Reply:
My apologies for the confusion. I mistakenly performed unnecessary type narrowing, so I removed it in commit https://github.com/jsx-eslint/eslint-plugin-react/commit/d59ac204a929314b6c24956f7c3b31dd036a28fa. Since the existing test cases seem to be unaffected, I did not add any additional tests.